metadetect/masking.py

```diff
@@ -3,6 +3,55 @@
 from .interpolate import interpolate_image_at_mask
 
 
+def apply_apodization_corrections(*, mbobs, ap_rad, mask_bit_val):
+    """Apply an apodization mask around the edge of the images in an mbobs to
+    prevent FFT artifacts.
+
+    Parameters
+    ----------
+    mbobs: ngmix.MultiBandObsList
+        The observations to mask
+    mask_bit_val: int
+        The bit to set in the bit mask for areas that are apodized.
+    ap_rad: float
+        When apodizing, the scale of the kernel. The total kernel goes from 0 to 1
+        over 6*ap_rad.
+    """
+    ap_mask = np.ones_like(mbobs[0][0].image)
+    _build_square_apodization_mask(ap_rad, ap_mask)
+
+    msk = np.where(ap_mask < 1)
+    if msk[0].size > 0:
+        for obslist in mbobs:
+            for obs in obslist:
+                # the pixels list will be reset upon exiting
+                with obs.writeable():
+                    obs.image *= ap_mask
```

Comment:
this could also be obs.image[msk] *= ap_msk[msk], but I understand if you want to move on

Reply:
done!